src/concepts/services/fetch.md

```diff
@@ -0,0 +1,147 @@
+{% endhint %}
+
+```rust 
+use yew::prelude::*;
+use yew::services::fetch::{FetchService, FetchTask};
+use serde::Deserialize;
+
+#[derive(Debug, Clone)]
+struct FetchServiceExample {
+    fetch_service: FetchService,
+    ft: Option<FetchTask>,
+    iss: Option<ISS>,
+    link: ComponentLink<Self>
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct ISSPosition {
+    latitude: String,
+    longitude: String
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct ISS {
+    message: String,
+    timestamp: i32,
+    iss_position: ISSPosition,
+    fetching: bool
+}
+
+#[derive(Debug, Clone)]
+enum Msg {
+    GetLocation,
+    Noop,
+    ReceiveLocation(ISS)
+}
+
+impl Component for FetchServiceExample {
+    type Message = Msg;
+    type Properties = ();
+    fn create(_: Self::Properties, link: ComponentLink<Self>) -> Self {
+        Self {
+            fetch_service: FetchService::new(),
+            ft: None,
+            iss: None,
+            link,
+            fetching: false
+        }
+    }
+    fn change(&mut self, _: Self::Properties) -> bool {
+        unimplemented!()
+    }
+    fn update(&mut self, msg: Self::Message) -> bool {
+        match msg {
+            GetLocation => {
+                // 1. build the request
+                let request = Request::get("http://api.open-notify.org/iss-now.json")
+                    .body(Nothing)
+                    .expect("Could not build request.");
+                // 2. construct a callback
+                let callback = self.link.callback(|response: Response<Json<Result<ISS, anyhow::Error>>>| {
+                    // split up the response into the HTTP data about the request result and data from the request
+                    let (meta, Json(data)) = response.into_parts();
+                    if meta.status.is_success() {
+                        match data.message {
+                            "success" => {
+                                Self::Message::ReceiveLocation(data.clone())
+                            }
+                            _ => {
+                                Self::Message::Noop
+                            }
+                        }
+                    } else {
+                        Self::Message::Noop
+                    }
+                });
+                // 3. pass the request and callback to the fetch service 
+                self.fetch_service.fetch(request, callback);
+                self.fetching = true;
+                // we want to redraw so that the page displays a 'fetching...' message to the user
+                true
+            }
+            Self::Message::ReceiveLocation(location) => {
+                self.iss = location;
+                self.fetching = false;
+                // we want to redraw so that the page no longer says 'fetching...'
+                true
+            }
+            _ => false
+        }
+    }
+    fn view(&self) -> Html {
+        html! {
+            <>
+                {if self.fetching {
```

Comment:
Can you please break out this block and the `match self.iss` block into their own methods on `FetchServiceExample`? I'd like the examples to set a good precedent for splitting up view logic

Reply:
Done (I think)!